packages/openapi3/src/schema-emitter-3-0.ts

```diff
@@ -206,12 +207,19 @@ export class OpenAPI3SchemaEmitter extends OpenAPI3SchemaEmitterBase<OpenAPI3Sch
           (schema instanceof Placeholder || "$ref" in schema) &&
           !(type && shouldInline(program, type))
         ) {
-          if (type && (type.kind === "Model" || type.kind === "Scalar")) {
+          if (type && type.kind === "Model") {
             return new ObjectBuilder({
               type: "object",
               allOf: Builders.array([schema]),
               ...additionalProps,
             });
+          } else if (type && type.kind === "Scalar") {
+            const stdName = $.scalar.getStdBase(type)?.name;
```

Comment:
I think this will break for anything else but string

Reply:
[Example playground](https://cadlplayground.z22.web.core.windows.net/prs/6506/?c=bW9kZWwgRm9vIHsKICBpMzI6xAUgfCBudWxsOwp9CgoKc2NhbGFyxRdleHRlbmRzIGludDMyOw%3D%3D&e=%40typespec%2Fopenapi3&options=%7B%7D)

Call `getSchemaForStdScalars` after getting the base to get the type + format